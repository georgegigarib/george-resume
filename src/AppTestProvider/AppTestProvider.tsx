import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
