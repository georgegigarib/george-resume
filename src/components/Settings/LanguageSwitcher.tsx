import { Menu, MenuItem, Button } from '@mui/material';
import { useState } from 'react';
import ChangeLanguageIcon from '@mui/icons-material/Language';
import ArrowDropDown from '@mui/icons-material/ArrowDropDown';
import { useLanguage } from '../../hooks/useLanguage';
import i18n from '../../../i18n.config';

interface LanguageOption {
  code: string;
  name: string;
}

const LanguageSwitcher = () => {
  const { changeLanguage } = useLanguage();
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);

  const languageOptions: LanguageOption[] = [
    { code: 'en', name: 'English' },
    { code: 'es', name: 'Español' },
  ];

  return (
    <>
      <div className="flex items-center justify-between w-full gap-10">
        <Button
          variant="contained"
          onClick={(event) => setAnchorEl(event.currentTarget)}
          style={{
            borderRadius: '20px',
            display: 'flex',
            alignItems: 'center',
          }}
          size='small'
        >
          <ChangeLanguageIcon style={{ marginRight: '8px' }} />
          {languageOptions.find((lang) => lang.code === i18n.language)?.name || 'English'}
          <ArrowDropDown style={{ marginLeft: 'auto', transform: anchorEl ? 'rotate(180deg)' : 'rotate(0deg)', transition: 'transform 0.3s ease' }} />
        </Button>

        <Menu
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={() => setAnchorEl(null)}
          PaperProps={{
            style: {
              boxShadow: '0px 4px 10px rgba(0, 0, 0, 0.4)',
              border: '1px solid rgba(0, 0, 0, 0.1)',
              borderRadius: '5px',
            },
          }}
        >
          {languageOptions.map((lang) => (
            <MenuItem 
              key={lang.code} 
              onClick={() => {
                changeLanguage(lang.code);
                setAnchorEl(null); // Cierra el menú después de seleccionar
              }}
            >
              {lang.name}
            </MenuItem>
          ))}
        </Menu>
      </div>
    </>
  );
};

export default LanguageSwitcher;
