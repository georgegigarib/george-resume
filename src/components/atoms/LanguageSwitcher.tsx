import { Menu, MenuItem, Button } from '@mui/material'
import { useState } from 'react'
import ChangeLanguageIcon from '@mui/icons-material/Language'
import ArrowDropDown from '@mui/icons-material/ArrowDropDown'
import { useLanguage } from '@/hooks/useLanguage'
import { languages, LanguageOption } from '@/locales/languages'
import i18n from '@/locales/i18n.config'

const LanguageSwitcher = () => {
  const { changeLanguage } = useLanguage()
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)

  const languageOptions: LanguageOption[] = Object.values(languages)

  const currentLanguage = languages[i18n.language as keyof typeof languages]

  return (
    <>
      <div
        className="flex items-center justify-between w-full gap-10"
        data-testid="language-switcher"
      >
        <Button
          variant="contained"
          onClick={(event) => setAnchorEl(event.currentTarget)}
          style={{
            borderRadius: '20px',
            display: 'flex',
            alignItems: 'center'
          }}
          size="small"
          data-testid="language-switcher-button"
        >
          <ChangeLanguageIcon style={{ marginRight: '8px' }} />
          {currentLanguage?.name || 'English'}
          <ArrowDropDown
            style={{
              marginLeft: 'auto',
              transform: anchorEl ? 'rotate(180deg)' : 'rotate(0deg)',
              transition: 'transform 0.3s ease'
            }}
          />
        </Button>

        <Menu
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={() => setAnchorEl(null)}
          PaperProps={{
            style: {
              boxShadow: '0px 4px 10px rgba(0, 0, 0, 0.4)',
              border: '1px solid rgba(0, 0, 0, 0.1)',
              borderRadius: '5px'
            }
          }}
          data-testid="language-menu"
        >
          {languageOptions.map((lang) => (
            <MenuItem
              key={lang.code}
              onClick={() => {
                changeLanguage(lang.code)
                setAnchorEl(null)
              }}
              data-testid={`language-option-${lang.code}`}
            >
              {lang.name}
            </MenuItem>
          ))}
        </Menu>
      </div>
    </>
  )
}

export default LanguageSwitcher
