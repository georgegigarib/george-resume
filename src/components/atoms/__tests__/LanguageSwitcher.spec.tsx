import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import { render, fireEvent } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import LanguageSwitcher from '@/components/atoms/LanguageSwitcher'

const changeLanguageMock = vi.fn()

vi.mock('@/hooks/useLanguage', () => ({
  useLanguage: () => ({
    changeLanguage: changeLanguageMock,
  }),
}))

describe('LanguageSwitcher component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <LanguageSwitcher />
      </AppTestProvider>
    )

  it('matches the snapshot', () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it('opens the menu when button is clicked', () => {
    const { getByTestId } = renderComponent()

    const button = getByTestId('language-switcher-button')
    fireEvent.click(button)

    expect(getByTestId('language-menu')).toBeTruthy()
    expect(getByTestId('language-option-en')).toBeTruthy()
    expect(getByTestId('language-option-es')).toBeTruthy()
  })

  it('changes language when an option is selected', () => {
    const { getByTestId } = renderComponent()

    const button = getByTestId('language-switcher-button')
    fireEvent.click(button)

    const spanishOption = getByTestId('language-option-es')
    fireEvent.click(spanishOption)

    expect(changeLanguageMock).toHaveBeenCalledWith('es')
  })
})
