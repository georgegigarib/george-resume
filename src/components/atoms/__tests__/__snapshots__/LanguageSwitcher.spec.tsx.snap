// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LanguageSwitcher component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between w-full gap-10"
    data-testid="language-switcher"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-6wsdc3-MuiButtonBase-root-MuiButton-root"
      data-testid="language-switcher-button"
      style="border-radius: 20px; display: flex; align-items: center;"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="LanguageIcon"
        focusable="false"
        style="margin-right: 8px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
        />
      </svg>
      English
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        style="margin-left: auto; transform: rotate(0deg); transition: transform 0.3s ease;"
        viewBox="0 0 24 24"
      >
        <path
          d="m7 10 5 5 5-5z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
